envs/monkey_zoo/blackbox/utils/gcp_machine_handlers.py

```diff
@@ -1,76 +1,83 @@
-        absolute_key_path = os.path.join(file_dir, GCPHandler.RELATIVE_KEY_PATH)
-        absolute_key_path = os.path.realpath(absolute_key_path)
-
-        if not os.path.isfile(absolute_key_path):
-            raise FileNotFoundError(
-                "GCP key not found. " "Add a service key to envs/monkey_zoo/gcp_keys/gcp_key.json"
-            )
-        return absolute_key_path
-
-    def start_machines(self, machine_list):
-        """
-        Start all the machines in the list.
-        :param machine_list: A space-separated string with all the machine names. Example:
-        start_machines(`" ".join(["elastic-3", "mssql-16"])`)
-        """
-        LOGGER.info("Setting up all GCP machines...")
-        try:
-            subprocess.call(  # noqa: DUO116
-                (GCPHandler.MACHINE_STARTING_COMMAND % (machine_list, self.zone)), shell=True
-            )
+# Key path location relative to this file's directory
+RELATIVE_KEY_PATH = "../../gcp_keys/gcp_key.json"
+DEFAULT_PROJECT = "guardicore-22050661"
+
+
+def initialize_gcp_client():
+    abs_key_path = get_absolute_key_path()
+
+    subprocess.call(get_auth_command(abs_key_path), shell=True)  # noqa: DUO116
+    LOGGER.info("GCP Handler passed key")
+
+    subprocess.call(get_set_project_command(DEFAULT_PROJECT), shell=True)  # noqa: DUO116
+    LOGGER.info("GCP Handler set project")
+    LOGGER.info("GCP Handler initialized successfully")
+
+
+def get_absolute_key_path() -> str:
+    file_dir = os.path.dirname(os.path.realpath(__file__))
+    absolute_key_path = os.path.join(file_dir, RELATIVE_KEY_PATH)
+    absolute_key_path = os.path.realpath(absolute_key_path)
+
+    if not os.path.isfile(absolute_key_path):
+        raise FileNotFoundError(
+            "GCP key not found. " "Add a service key to envs/monkey_zoo/gcp_keys/gcp_key.json"
+        )
+    return absolute_key_path
+
+
+def start_machines(machine_list):
+    """
+    Start all the machines in the list.
+    :param machine_list: A dictionary with zone and machines per zone.
+    """
+    LOGGER.info("Setting up all GCP machines...")
+    try:
+        arglist = []
+        for zone in machine_list:
+            arglist.append((MACHINE_STARTING_COMMAND, machine_list, zone))
+        with Pool(2) as pool:
+            pool.map(run_gcp_command, arglist)
             LOGGER.info("GCP machines successfully started.")
-        except Exception as e:
-            LOGGER.error("GCP Handler failed to start GCP machines: %s" % e)
-
-    def stop_machines(self, machine_list):
-        try:
-            subprocess.call(  # noqa: DUO116
-                (GCPHandler.MACHINE_STOPPING_COMMAND % (machine_list, self.zone)), shell=True
-            )
+    except Exception as e:
+        LOGGER.error("GCP Handler failed to start GCP machines: %s" % e)
+
+
+def stop_machines(machine_list):
+    try:
+        arglist = []
+        for zone in machine_list:
+            arglist.append((MACHINE_STOPPING_COMMAND, machine_list, zone))
```

Comment:
What about using list a compehension here? Can we remove the duplication between the start and stop machines functions?